### ISSUE TITLE
Dynamic axes incorrectly traced in nn.Embedding when exporting to ONNX

### ISSUE THREAD

**Issue by taraspiotr** [pytorch/pytorch#70110]
### 🐛 Describe the bug

```
import torch
import torch.onnx
import onnx

module = torch.nn.Embedding(20, 40)
inputs = torch.zeros(2, dtype=torch.int)
torch.onnx.export(module, (inputs, ), "model.onnx", input_names=["ids"],
                  dynamic_axes={"ids": {0: "batch"}}
                 )
onxx_model = onnx.load("model.onnx")
print(onxx_model.graph.output)
```
Output:
```
[name: "2"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "Gather2_dim_0"
      }
      dim {
        dim_param: "Gather2_dim_1"
      }
    }
  }
}
]
```
The last dimension should be a constant (40, embedding dimension), but when the first (batch) dimension is set to be dynamic onnx intereprets the second dim as dynamic as well.

### Versions

```
Collecting environment information...
PyTorch version: 1.11.0a0+b6df043
Is debug build: False
CUDA used to build PyTorch: 11.5
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.21.3
Libc version: glibc-2.31

Python version: 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:59:51)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.11.0-40-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.5.50
GPU models and configuration: 
GPU 0: Quadro RTX 6000
GPU 1: NVIDIA GeForce GT 710

Nvidia driver version: 470.82.00
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.3.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.3.0
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.4
[pip3] nvidia-dlprof-pytorch-nvtx==1.7.0
[pip3] pytorch-quantization==2.1.2
[pip3] torch==1.11.0a0+b6df043
[pip3] torch-tensorrt==1.0.0a0
[pip3] torchtext==0.12.0a0
[pip3] torchvision==0.11.0a0
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.5                      281    conda-forge
[conda] mkl-include               2019.5                      281    conda-forge
[conda] numpy                     1.21.4           py38he2449b9_0    conda-forge
[conda] nvidia-dlprof-pytorch-nvtx 1.7.0                    pypi_0    pypi
[conda] pytorch-quantization      2.1.2                    pypi_0    pypi
[conda] torch                     1.11.0a0+b6df043          pypi_0    pypi
[conda] torch-tensorrt            1.0.0a0                  pypi_0    pypi
[conda] torchtext                 0.12.0a0                 pypi_0    pypi
[conda] torchvision               0.11.0a0                 pypi_0    pypi

```

**Comment by fatcat-z** [comment]
This is being tracked internally by Microsoft at https://msdata.visualstudio.com/Vienna/_workitems/edit/1567951.

**Comment by AllenTiTaiWang** [comment]
Hi (tag relevant users),

We’ve gone ahead and closed this issue because it has been resolved elsewhere. This resolution can be found with using nightly torch.
If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.

Thanks,
ONNX Converter team
